### PR TITLE
feat: Get rid of Axios, added api folder for requests, hooks use requ…

### PR DESCRIPTION
**Pull request by nosensus**
Git rid of use Axios https client and use native react fetch

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -13,8 +13,8 @@ function App() {
       <Header />
       <Routes>
         <Route path="/" element={<TodoItemsActivePage />} />
-        <Route path="/all" element={<TodoItemsPage />} />
-        <Route path="/complete" element={<TodoItemsCompletePage />} />
+        {/* <Route path="/all" element={<TodoItemsPage />} />
+        <Route path="/complete" element={<TodoItemsCompletePage />} /> */}
```
Comment (adizka) [inline src/App.tsx]: Why did you comment these?

File: src/hooks/useTodoItemDelete.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  const todoItemDelete = async (id: string) => {
+    setQuery({ ...query, isLoading: true });
+
+    try {
+      await deleteTodoItem(id);
+    } catch (error) {
+
+    } finally {
+
```
Comment (adizka) [inline src/hooks/useTodoItemDelete.ts]: Loading has to be toggled.

File: src/hooks/useTodoItemEdit.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    todoItemCard.isImportant = isImportant;
+    todoItemCard.category = +todoItemCard.category;
+    todoItemCard.cardColor = +todoItemCard.cardColor;
+
+    try {
+      editTodoItem(todoItemCard);
+    } catch (error) {
+
+    } finally {
+
```
Comment (adizka) [inline src/hooks/useTodoItemEdit.ts]: Loading has to be placed false

File: src/pages/TodoItemsCompletePage.tsx [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
+  //                 onCardDelete={() => undefined}
+  //                 onCardEdit={() => undefined}
+  //               />
+  //             ))}
+  //         </div>
+  //       }
+  //     </section>
+  //   </>
+  // );
 };
```
Comment (adizka) [inline src/pages/TodoItemsCompletePage.tsx]: Why it was commented?

**Review by adizka — APPROVED** [review]
check comments

File: src/hooks/useTodoItemDelete.ts (hunk 2 above)
Comment (nosensus) [reply src/hooks/useTodoItemDelete.ts]: Definitely yes.

File: src/pages/TodoItemsCompletePage.tsx (hunk 4 above)
Comment (nosensus) [reply src/pages/TodoItemsCompletePage.tsx]: I have no clue why todo list does not show on pages. The reason was in re-calling **setQuery** in the try catch construction. Fixed. But forget to uncomment.

**Pull request merged**